core/arch/arm32/kernel/thread.c

```diff
@@ -633,6 +655,7 @@ void thread_init_primary(const struct thread_handlers *handlers)
 	 * checks verifies that the offsets used in assembly code matches
 	 * what's used in C code.
 	 */
+#ifdef ARM32
```

Comment:
This is OK as it is now. My comment was relevant if a _arm32.c file was created. I guess that it is no more relevant.